lib/elixir/lib/calendar.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Calendar do
+  @moduledoc """
+  This module defines the responsibilities for working with
+  calendars, dates, times and datetimes in Elixir.
+
+  Currently it defines types but may define the calendar
+  behaviour in future versions. For the actual date, time
+  and datetime structures, see `Date`, `Time`, `NaiveDateTime`
+  and `DateTime`.
+
+  Note the year, month, day, etc designations are over specified
+  (i.e. an integer instead of 1..12 for months) because different
+  calendars may have a different number of numbers and so on.
+  """
+
+  @type year        :: integer
+  @type month       :: integer
+  @type day         :: integer
+  @type hour        :: integer
+  @type minute      :: integer
+  @type second      :: integer
+  @type microsecond :: integer
+
+  @typedoc "A calendar implementation"
+  @type calendar    :: module
+
+  @typedoc "The timezone ID according to Unicode's CLDR"
+  @type time_zone   :: String.t
+
+  @typedoc "The timezone abbreviation"
+  @type zone_abbr   :: String.t
+
+  @typedoc "The timezone UTC offset in seconds"
+  @type utc_offset  :: integer
+
+  @typedoc "The timezone standard offset in seconds (not zero in summer times)"
+  @type std_offset  :: integer
+end
+
+defmodule Date do
+  @moduledoc """
+  A date implementation.
+  """
+  defstruct [:year, :month, :day]
```

Comment:
A Date should also have a calendar field. Calendars are more about dates than they are about datetimes, though there are often assumptions made about time for the sake of conversion (such as making the assumption that the epoch for all calendars begins on the first day containing noon of that year).